python/ray/dataframe/dataframe.py

```diff
@@ -776,9 +776,20 @@ def dropna(self, axis=0, how='any', thresh=None, subset=None,
             DataFrame with the dropna applied.
         """
         if is_list_like(axis):
-            raise NotImplementedError(
-                "To contribute to Pandas on Ray, please visit "
-                "github.com/ray-project/ray.")
+            axis = set([pd.DataFrame()._get_axis_number(ax) for ax in axis])
+            result = self
+            for ax in axis:
+                result = result.dropna(
+                    axis=ax, how=how, thresh=thresh, subset=subset)
+            if not inplace:
+                return result
+
+            return self._update_inplace(
```

Comment:
Pass `_block_partitions` in instead, it's more efficient.

Reply:
done!